### PR TITLE
Handling XML faults

### PR DESCRIPTION
**Pull request by mbojan**
Addresses #24 .

In short the flow is to:

- try to parse with `read_xml`
- if that fails, try `read_html`
- `resumptionToken` is extracted in both cases above
- if `as="raw"` proceed as usual in both cases
- if `as != "raw"` save raw XML and proceed with the token (if any)

It is getting complicated and I think we should use some package options. In this PR I have already used one in `read_xml_safely`. We might use `options` or have some object stored in package namespace (like in knitr `opts_chunk` etc.)

At this moment this is used only in `while_oai` and functions that use it. `read_xml_safely` could be called  instead of `read_xml` in other functions too.

### CODE REVIEW

File: R/read_xml_safely.R [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+
+
+
+# Read XML safely
+#
+# Removes illegal characters
+# @param invalid_as NA or character. If not NA, replace invalid characters with `invalid_as`
+read_xml_safely <- function(x, ..., xml_invalid_as=getOption("oai.xml_invalid_as", "") ) {
```
Comment (sckott) [inline R/read_xml_safely.R]: @mbojan where does the `oai.xml_invalid_as` option come from? that is, it's fine to have it not be set, and user can optionally set it?

File: R/while_oai.R [hunk 2]
```diff
@@ -16,26 +16,41 @@ while_oai <- function(url, args, token, as, dumper=NULL, dumper_args=NULL, ...)
...
     tt <- content(res, "text")
-    xml_orig <- xml2::read_xml(tt)
-    handle_errors(xml_orig)
-    xml <- xml2::xml_children(xml2::xml_children(xml_orig)[[3]])
-    trytok <- xml2::as_list(xml[sapply(xml, xml_name) == "resumptionToken"])
-    if (length(trytok) == 0) {
-      tok <- list(token = "")
+
+    # try parsing
+    parsed <- try( read_xml_safely(tt), silent=TRUE )
```
Comment (sckott) [inline R/while_oai.R]: do you intend for user to be able to pass  `xml_invalid_as` and any options through `...`?

**Comment by sckott** [comment]
> It is getting complicated and I think we should use some package options.

What do you mean by this?  New issue to discuss?

**Comment by sckott** [comment]
Passing now, i'll merge after you address above questions

File: R/read_xml_safely.R (hunk 1 above)
Comment (mbojan) [inline R/read_xml_safely.R]: Yes. There is a default value which can be changed by setting the option.

File: R/while_oai.R (hunk 2 above)
Comment (mbojan) [inline R/while_oai.R]: I wanted to avoid passing arguments. User can set an `option` instead.

**Comment by mbojan** [comment]
>> It is getting complicated and I think we should use some package options.

> What do you mean by this? New issue to discuss?

Sorry. I should have probably make it a separate issue or not mention altogether. Yes, I was wondering out loud whether we should have package options... Maybe it is not a real issue now. I just used two options in this PR.

**Pull request merged**